linera-core/src/client/mod.rs

```diff
@@ -365,25 +365,27 @@ where
         &self,
         remote_node: &RemoteNode<impl ValidatorNode>,
         chain_id: ChainId,
-        certificates: Vec<Certificate>,
+        certificates: Vec<ConfirmedBlockCertificate>,
     ) -> Option<Box<ChainInfo>> {
         let mut info = None;
         for certificate in certificates {
             let hash = certificate.hash();
-            if !certificate.value().is_confirmed() || certificate.value().chain_id() != chain_id {
+            if certificate.executed_block().block.chain_id != chain_id {
                 // The certificate is not as expected. Give up.
                 warn!("Failed to process network certificate {}", hash);
                 return info;
             }
-            let mut result = self.handle_certificate(certificate.clone(), vec![]).await;
+            let mut result = self
+                .handle_certificate(certificate.clone().into(), vec![])
```

Comment:
`handle_certificate` expects the "old" `Certificate` type still. Internally it pattern-matches on the `value` type (`ConfirmedValue` enum) and forwards to the specialized handlers.